Makefile

```diff
@@ -190,6 +190,13 @@ clint: build/.ran-cmake
 clint-full: build/.ran-cmake
 	+$(BUILD_CMD) -C build clint-full
 
+clang-tidy: generate-compilation-db
+	bash scripts/run_clang_tidy.sh $(FILE)
```

Comment:
is the run_clang_tidy.sh script really needed? it's not a pattern we usually follow. Better not to spread things out if possible, also keeps the Makefile more portable (e.g. on Windows)

Reply:
Only if you want a way to run it per-file from the CLI. If you're running `clangd` then it can also manage `clang-tidy` for you and dynamically display any warnings and code-actions through LSP.

Alternatively, we can introduce another target called  `clang-tidy-file` or something, which would allow us to skip validating the variable inside make.

I didn't want to do conditional run from inside make, because that would get hacky very quickly: https://stackoverflow.com/a/45003119

Reply:
I think this could be done without much hassle inside the makefile. Something like the following should work:

```
FILE ?= $(shell git ls-files *.c)
clang-tidy: generate-compilation-db
	  clang-tidy --config-file .clang-tidy -p build $(FILE)
```

So if it's run as `make clang-tidy` then all files will be used. If it's run with `make clang-tidy FILE=src/nvim/screen.c` then it's only used on `screen.c`.